### PR TITLE
Small Tweaks to CMake build

### PR DESCRIPTION
**Pull request by mgaudet**
Mostly derived from hooking up a consuming compiler project.

### CODE REVIEW

File: cmake/compiler_support.cmake [hunk 1]
```diff
@@ -302,8 +302,17 @@ function(create_omr_compiler_library)
       list(REMOVE_ITEM CORE_COMPILER_OBJECTS ${abs_filename})
    endforeach()
 
+   # Convert pasm files ot asm files: run this before masm2gas to ensure
```
Comment (Leonardo2718) [inline cmake/compiler_support.cmake]: typo: "ot" -> "to".

File: cmake/compiler_support.cmake [hunk 2]
```diff
@@ -302,8 +302,17 @@ function(create_omr_compiler_library)
       list(REMOVE_ITEM CORE_COMPILER_OBJECTS ${abs_filename})
    endforeach()
 
+   # Convert pasm files to asm files: run this before masm2gas to ensure 
+   # asm files subsequently get picked up. 
+   pasm2asm_files    (CORE_COMPILER_OBJECTS ${COMPILER_NAME} ${CORE_COMPILER_OBJECTS})
```
Comment (charliegracie) [inline cmake/compiler_support.cmake]: Why is there extra spaces between the function name and the parameters?

Comment (mgaudet) [inline cmake/compiler_support.cmake]: Removed; they were there to align with the below to allow me to visually ensure the parallel with masm2gas.

**Comment by mgaudet** [comment]
Awkard note: 

Because 87221f6 is marked as `[ci skip]`, and is the topmost commit, apparently our CI services have declined to build this PR now. 

I'm going to rebase this branch, squashing 87221f6 into 0647279 to ensure this branch gets built.

File: cmake/platform.cmake [hunk 3]
```diff
@@ -31,18 +31,18 @@ message(STATUS "CMAKE_CROSSCOMPILING=${CMAKE_CROSSCOMPILING}")
 omr_detect_system_information()
 
 # Pickup OS info 
-include(cmake/platform/os/${OMR_HOST_OS}.cmake OPTIONAL)
+include(${OMR_ROOT}/cmake/platform/os/${OMR_HOST_OS}.cmake OPTIONAL)
```
Comment (rwy0717) [inline cmake/platform.cmake]: Can we make this:
```cmake
include(./platform/os/${OMR_HOST_OS}.cmake OPTIONAL)
```
 instead? Same for arg and tools.

Comment (mgaudet) [inline cmake/platform.cmake]: Seems no? Not sure why, but in the consuming project that inspired this work that still fails to find the include files.

Comment (rwy0717) [inline cmake/platform.cmake]: Sadness, but OK.

**Comment by rwy0717** [comment]
LGTM

File: cmake/platform.cmake [hunk 4]
```diff
@@ -60,11 +60,13 @@ omr_add_prefix(OMR_ARCH_DEFINITIONS_PREFIXED ${OMR_C_DEFINITION_PREFIX} ${OMR_AR
 
 
 add_definitions(
-	${OMR_OS_COMPILE_OPTIONS}
 	${OMR_OS_DEFINITIONS_PREFIXED}
 	${OMR_ARCH_DEFINITIONS_PREFIXED}
 )
 
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} ${OMR_OS_COMPILE_OPTIONS})
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} ${OMR_OS_COMPILE_OPTIONS})
```
Comment (rwy0717) [inline cmake/platform.cmake]: This needs to be stringified properly, right?

Comment (mgaudet) [inline cmake/platform.cmake]: In my experience, no. However, I don't have a strong intuition for when stringification  is required in CMake.

Comment (rwy0717) [inline cmake/platform.cmake]: My understanding is: `CMAKE_<LANG>_FLAGS` is a string of space separated flags. Appending a list this way will insert unwanted semicolons into the compile command. You can check out how we're already appending flags just [below](https://github.com/eclipse/omr/pull/1496/files#diff-9086bbfa8e248778f183939aa77c67a8R72). This PR passed CI, but TBH I don't know why.

Comment (mgaudet) [inline cmake/platform.cmake]: Alright, fixed.

**Comment by rwy0717** [comment]
Even still, since `OMR_OS_COMPILE_OPTIONS` is a list, we will need to transform the semicolon separators into spaces. This will work for now, since we have exactly 1 option. I believe we should continue to use lists to track options, so I've written some tiny utility functions which should make it easier to handle. I'll open a PR soon.

**Comment by rwy0717** [comment]
OK, could be some misinformation on my part, that's actually a relief :)

**Comment by mgaudet** [comment]
So, do we want to pull 26fc349? It seems that CMake's handling of list ↔️ string conversion is more robust than we worry...

**Comment by charliegracie** [comment]
I have only tested with `26fc349` so I am not sure what it would do without it and I will not have time to test it today.  I am ok with accepting this and we can modify in the future if it does not need to be there.

**Comment by mgaudet** [comment]
👍

**Comment by rwy0717** [comment]
In the future, maybe we want to use `add_compile_options()` (wherever possible), and sidestep the whole list/string issue.

**Comment by charliegracie** [comment]
Does add compile options add to CFLAGS and CPPFLAGS and ASFLAGS?

**Comment by rwy0717** [comment]
Yes I believe so, which does make it harder (or impossible) to use in some cases.

**Comment by rwy0717** [comment]
As an FYI, I checked out 3d5dff9, the commit before proper stringification. I ran cmake with the Gnu Makefile generator. The extra semicolons were present in the compile line.  `make VERBOSE=1` failed immediately with:

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f tools/tracegen/CMakeFiles/trace.dir/build.make tools/tracegen/CMakeFiles/trace.dir/build
[  1%] Building CXX object tools/tracegen/CMakeFiles/trace.dir/ArgParser.cpp.o
cd /Users/rwyoung/wsp/omr/build/tools/tracegen && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DJ9HAMMER -DOSX -DUT_DIRECT_TRACE_REGISTRATION -D_FILE_OFFSET_BITS=64 -I/Users/rwyoung/wsp/omr/build/tools/tracegen -I/Users/rwyoung/wsp/omr/tools/tracegen -I/Users/rwyoung/wsp/omr/build -I/Users/rwyoung/wsp/omr/./include -I/Users/rwyoung/wsp/omr/./include_core -I/Users/rwyoung/wsp/omr/./third_party  -O0 -g3 -fno-exceptions -fno-rtti;-pthread   -o CMakeFiles/trace.dir/ArgParser.cpp.o -c /Users/rwyoung/wsp/omr/tools/tracegen/ArgParser.cpp
clang: error: no input files
/bin/sh: -pthread: command not found
make[2]: *** [tools/tracegen/CMakeFiles/trace.dir/ArgParser.cpp.o] Error 127
make[1]: *** [tools/tracegen/CMakeFiles/trace.dir/all] Error 2
make: *** [all] Error 2
```

Maybe it's the ninja-generator fixing things up?

**Comment by mgaudet** [comment]
Ooof. Definitely furthers the argument that we ought to have Makefile based builds in addition to our Ninja builds.

**Comment by rwy0717** [comment]
I went ahead and opened the cmake utilities PR I mentioned above: https://github.com/eclipse/omr/pull/1507

**Pull request merged**